0002-crypto-conditions/0002-crypto-conditions.md

```diff
@@ -110,387 +110,380 @@ informative:
 
-### Example Condition
+  1. Assemble the fingerprint content and if neccessary calculate the hash digest of this data.
+  2. Calculate the maximum fulfillment length of a fulfillment
+  
+For compound types the fingerprint content will contain the complete, encoded, condition for all sub-crypto-conditions. Implementations MUST abide by the ordering rules provided when assembling the fingerprint content.
 
-An example condition in string format:
+When calculating the fingerprint of a compound crypto-condition implementations MUST first derive the condition for all sub-fulfillments and include these conditions when assembling the fingerprint content.
 
-    cc:0:3:dB-8fb14MdO75Brp_Pvh4d7ganckilrRl13RS_UmrXA:66
+## Conditions as Public Keys
 
-The example has the following attributes:
+Since the condition is just a fingerprint and meta-data about the crypto-condition it can be transmitted freely in the same way a public key is shared publicly. It's not possible to derive the fulfillment from the condition.
 
-| Field                | Value | Description |
-|----------------------|-------|----------------------------------------------|
-| preface              | `cc`  | Constant. Indicates this is a condition. |
-| type                 | `0`   | Type 0 is [PREIMAGE-SHA-256][]. |
-| featuresBitmask      | `3`   | Boolean-OR combination of feature suites SHA-256 (feature bit 0x01) and PREIMAGE (feature bit 0x02). |
-| fingerprint          | `dB-8fb14MdO75Brp_Pvh4d7ganckilrRl13RS_UmrXA` | The hash of the fulfillment for this condition. |
-| maxFulfillmentLength | `66`  | The fulfillment payload is 66 bytes long, before being BASE64URL-encoded. |
+# Format
 
+A description of crypto-conditions is provided in this document using Abstract Syntax Notation One (ASN.1) as defined in [ITU.X680.2015](#itu.X680.2015).
 
-## Fulfillment {#fulfillment-format}
-Below are the string and binary encoding formats for a fulfillment.
+## Encoding Rules
 
-### String Format {#string-fulfillment-format}
+Implementations of this spec MUST support encoding and decoding using Octet Encoding Rules (OER) as defined in [ITU.X696.2015](#itu.X696.2015). This is the canonical encoding format. Where alternate encoding rules are defined and used these should be applied through to all ASN.1 defined data structures.
 
-Fulfillments are ASCII encoded as:
+For example, implementations that choose to use DER encoding should also use DER encoding when assembling the content of a condition fingerprint before it is hashed.
 
-    "cf:" BASE16(type) ":" BASE64URL(payload)
+## Condition {#condition-format}
 
-### Binary Format {#binary-fulfillment-format}
+The binary encoding of conditions is dependant on their type. The overall size of conditions is not fixed as some types (compound conditions) contain additional fields (subtypes) and the algorithm for deriving the fingerprint differs by type therefor the fingerprint size also differs.
```

Comment:
See my earlier comment. We probably want to change the section below to remove subtypes from simple types.